control/control.go

```diff
@@ -5,37 +5,86 @@ import (
 	"bytes"
 	"encoding/base64"
 	"encoding/json"
+	"errors"
 	"fmt"
+	"io"
 	"io/ioutil"
 	"log"
+	"net/http"
+	"net/url"
+	"os"
 	"path/filepath"
+	"strings"
 	"text/template"
+	"time"
 )
 
+// Logger is a STDERR logger
+var Logger = log.New(os.Stderr, "", 0)
+
 // Configuration is the overall data structure unmarshalled from JSON
 type Configuration struct {
 	Filters map[string]map[string]string
 	BaseDir string
 }
 
-// Read reads in the configuration and returns the object
-func (c *Configuration) Read(file string) error {
-	raw, err := ioutil.ReadFile(file)
+// Get fetches the control from a location and returns a io.ReadCloser
+func Get(location string, headers []string) (io.ReadCloser, error) {
+	u, err := url.ParseRequestURI(location)
```

Comment:
I assume this returns an error if location isn't a URI (e.g. a file path)?

Reply:
sort of, it returns an error for things that are invalid URIs... so a file path like `/foo/bar/baz` is _not_ an error, but that is handled by checking the `scheme` below.